### ISSUE TITLE
Names of parameters may benefit from not being abbreviated

### ISSUE THREAD

**Issue by alok** [williamFalcon/pytorch-lightning#119]
I know this is nitpicky, but I think good naming is worth a lot of thought.

A lot of the API seems unhelpfully abbreviated to me, especially since lighting is designed so that you don't have to handle manual details like dataloaders more than necessary.

Names like `tng_dataloader` don't seem to buy anything over `train_dataloader` or `training_dataloader` since they're written only once and read many more times. Really, `tng` could be replaced with `training` or `train` elsewhere too.

`data_batch` seems redundant, I think it could just be called `batch` since in context it can only represent data anyway, and `batch_nb` is already a separate argument.

**Describe the solution you'd like**

Rename. The library is still in early days.

**Comment by williamFalcon** [comment]
@alok sure, can you first list out the name changes here.

from -> to:

example:
```
tng_dataloader -> train_dataloader
```

**Comment by alok** [comment]
- `tng` -> `training`
- `pct` -> `percent` or `frac`
- `nb` -> `num` (this one i don't have as strong feelings about)
- `data_batch` -> `batch`
- `prog` -> `progress`
- `outputs` in `validation_end` -> `progress_metrics`
- `dataloader` -> `loader` (that it's a DataLoader is clear, so the `data` part is redundant, but also no really strong feeling)
- `current_epoch` -> `epoch` (the only mixup that could be had is with the total number of epochs, and that could be called something like `epochs`)
- `gradient_clip` -> `gradient_clip_val` (`gradient_clip` sounds like a boolean indicating whether to clip or not)
- `gpus` in trainer -> `gpu_ids`
- `add_log_row_interval` -> `row_log_interval`

**Comment by williamFalcon** [comment]
awesome.   

Let's do these:

```
tng -> training   
keep pct   
keep nb   
data_batch -> batch    
prog -> progress   
keep progress_metrics    
keep dataloader    
keep current_epoch    
gradient_clip -> graient_clip_val    
keep gpus
add_log_row_interval -> row_log_interval


```

**Comment by Ir1d** [comment]
I propose to rename `update_tng_log_metrics` to `update_train_log_metrics`

**Comment by williamFalcon** [comment]
@alok https://gitter.im/PyTorch-Lightning/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge

**Comment by alok** [comment]
I'll hold off on this until #146 is resolved, since it affects this.

**Comment by IvanLazarevsky** [comment]
> Let's do these:
> 
> ```
> gradient_clip -> gradient_clip_val    
> ```
I suggest gradient_clip_norm, because pytorch has torch.nn.utils.clip_grad_value_, which clips individual partial derivatives using torch.clamp and it would be confusing.

**Comment by williamFalcon** [comment]
Merged #124
